File: chtl/src/chtl/lexer/token.rs
#[derive(Debug, PartialEq, Clone)]
pub enum Token {
    // Literals
    Identifier(String),
    String(String),
    Number(f64),

    // Keywords
    Text,
    Style,
    Script,
    Template,
    Custom,
    Origin,
    Element,
    Import,
    Namespace,
    Configuration,
    Use,
    Html5,
    Inherit,
    Delete,
    Insert,
    From,
    As,
    Except,

    // Symbols
    Colon,
    Equal,
    LBrace,
    RBrace,
    LParen,
    RParen,
    LBracket,
    RBracket,
    Semicolon,
    Comma,
    Dot,
    At,
    Hash,
    Ampersand,
    Dollar,

    // Operators
    Plus,
    Minus,
    Star,
    Slash,
    Percent,
    Power,
    Question,
    LogicalAnd,
    LogicalOr,
    GreaterThan,
    LessThan,
    GreaterThanOrEqual,
    LessThanOrEqual,

    // Comments
    LineComment(String),
    BlockComment(String),
    GeneratorComment(String),

    // Special
    CHTLJSStart, // {{
    CHTJSEnd,    // }}

    // End of File
    EOF,
}

File: CHTL/CHTLJS/src/lexer/Token.cpp
#include "CHTLJS/include/lexer/Token.h"

namespace CHTLJS
{
    std::string TokenTypeToString(TokenType type)
    {
        switch (type)
        {
            case TokenType::RAW_JS:
                return "RAW_JS";
            case TokenType::END_OF_FILE:
                return "END_OF_FILE";
            case TokenType::LBRACE_BRACE:
                return "LBRACE_BRACE";
            case TokenType::RBRACE_BRACE:
                return "RBRACE_BRACE";
            case TokenType::LBRACE:
                return "LBRACE";
            case TokenType::RBRACE:
                return "RBRACE";
            case TokenType::LPAREN:
                return "LPAREN";
            case TokenType::RPAREN:
                return "RPAREN";
            case TokenType::LBRACKET:
                return "LBRACKET";
            case TokenType::RBRACKET:
                return "RBRACKET";
            case TokenType::COMMA:
                return "COMMA";
            case TokenType::COLON:
                return "COLON";
            case TokenType::SEMICOLON:
                return "SEMICOLON";
            case TokenType::ASSIGN:
                return "ASSIGN";
            case TokenType::IDENT:
                return "IDENT";
            case TokenType::ARROW:
                return "ARROW";
            case TokenType::BIND_OP:
                return "BIND_OP";
            default:
                return "UNKNOWN";
        }
    }

} // namespace CHTLJS
